addons/controllers/clusterbootstrap_controller.go

```diff
@@ -270,15 +270,21 @@ func (r *ClusterBootstrapReconciler) createOrPatchClusterBootstrapFromTemplate(c
 
 	clusterBootstrap := &runtanzuv1alpha3.ClusterBootstrap{}
 	err := r.Client.Get(r.context, client.ObjectKeyFromObject(cluster), clusterBootstrap)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	if clusterBootstrap.Spec != nil && clusterBootstrap.Spec.Paused {
```

Comment:
In predicates, we couldn't connect resources like Cluster CR to ClusterBootstrap.

Reply:
I figured it will be easier to let the mapping function do the job, and have the pause logic in reconciliation loop. We also did the same in addons manager.

Reply:
I think it can be done if the predicate function is a receiver of ClusterBootstrapReconciler instead of standalone but is probably little more code for the predicate function because need to link ClusterBootstrap from Cluster object.

The reason I like EventFilter because I can just look at `SetupWithManager` to understand what conditions need to be met instead of going through reconcile loop to check for conditions that need to be met. 

With that being said not a blocker and this works as well.